src/promptflow/tests/executor/unittests/_core/test_tools_manager.py

```diff
@@ -0,0 +1,88 @@
+from pathlib import Path
+
+import pytest
+
+from promptflow import tool
+from promptflow._core._errors import PackageToolNotFoundError
+from promptflow._core.tools_manager import ToolLoader
+from promptflow.contracts.flow import Node, ToolSource, ToolSourceType
+from promptflow.contracts.tool import Tool, ToolType
+from promptflow.exceptions import UserErrorException
+
+
+@pytest.mark.unittest
+class TestToolLoader:
+    def test_load_tool_for_node_with_invalid_node(self):
```

Comment:
Multi cases are tested in this method. Consider using pytest.parametrize.

Reply:
I have another pr to add unit tests for tool-related code, will fix in there.